### DISCUSSION TITLE
Re: Patching API what is it patching

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [JasperFx/marten#2079] [Q&A]
Re: the [Patching API](https://martendb.io/v3/documentation/documents/advanced/patch_api), if I understand it correctly, it is patching changes to domain model vis-a-vis what has been serialized to the database? How about more than perhaps minor changes, properties here and there, changes to _namespace paths_, etc. Could you elaborate please on some more, other, in depth, use cases? Thank you...

**Comment by mwpowellhtx** [comment]
Doing a bit of background scanning, FYI TL;DR, FWIW, is PLV8, V8, etc, the best possible path forward? [Seems perhaps not](https://github.com/plv8/plv8/issues/364), but I am not as intimately familiar with the inward facing consequences one way or another. Just a heads up.

**Reply by jeremydmiller** [reply]
All patching does is exactly what is documented in the page you linked to. Not sure if you were the same person asking about this on Gitter today, but long term it's unclear if PLv8 is viable or replaced by something else. 

It is possible to do at least a subset of what our patching API does today with native postgresql operators, and we'll likely pursue that at some point (it's been in the GitHub backlog for quite a while). As far as changing namespaces and such, you do also have the Javascript transforms that do anything.

Or not what you want to hear, but you can always drop down to raw SQL.

**Reply by mwpowellhtx** [reply]
Yep, fair enough; just sizing up the overall cost, opportunity, etc. Thanks for the feedback and clarification.
